src/textual/app.py

```diff
@@ -2977,11 +2979,14 @@ def notify(
             ```
         """
         notification = Notification(message, title, severity, timeout)
-        self._notifications.add(notification)
+        self.post_message(Notify(notification))
+
+    def _on_notify(self, event: Notify) -> None:
+        """Handle notification message."""
+        self._notifications.add(event.notification)
         self._refresh_notifications()
-        return notification
 
-    def unnotify(self, notification: Notification, refresh: bool = True) -> None:
+    def _unnotify(self, notification: Notification, refresh: bool = True) -> None:
```

Comment:
I've never used this API but this method seems like it's really useful if it's public?

Is this method how we discard notifications manually? It seems useful if notifications are longer-lived - to allow us to remove ones that are no longer relevant.

Reply:
The problem is not that `self.notify()` doesn't add the notification immediately. It posts a message, and returns `None`. That means that the user won't have a reference to the Notification object. And even if they did, it might not yet be active.

Programatically removing notifications is useful, but we will need another mechanism for that.

Reply:
Not saying it's for this PR, but I think it'd be nice if `notify` still returned a `Notification` as a handle to a notification which may or may not be active. Then, if you try to `unnotify` a `Notification` that isn't active then it'd just prevent it from appearing.